src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpoint.java

```diff
@@ -773,6 +773,31 @@ private CollectionExerciseDTO getCollectionExerciseDTO(
     return ResponseEntity.ok(result);
   }
 
+  @RequestMapping(value = "/events", method = RequestMethod.GET)
+  public ResponseEntity<HashMap<UUID, List<EventDTO>>> getMultipleCollectionExerciseEvents(
+      @RequestParam("ids") final List<UUID> ids) throws CTPException {
+    List<EventDTO> result =
```

Comment:
I think this can be simplified. The for loop looks like a groupingBy operation. Would have to change the method signature to return a Map instead of a HashMap though.
```
Map<UUID, List<EventDTO>> response =
        this.eventService
            .getEvents(ids)
            .stream()
            .map(EventService::createEventDTOFromEvent)
            .collect(Collectors.groupingBy(EventDTO::getCollectionExerciseId,
                Collectors.toList()));

    return ResponseEntity.ok(response);
```